[Ne Zhijian, Didymus] iP

didymusne opened the pull request:
### Duke
> "The all-in-one personal assistant chatbot that can help you manage your tasks" - Didymus ([source](https://github.com/didymental/ip))

Duke frees your mind of having to remember things you need to do. It's, 
* text-based
* easy to learn
* ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. Download it from [here](https://github.com/didymental/ip/releases/tag/A-Jar)
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you :smiling_face_with_three_hearts:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you are a Java programmer, you can use it to practise Java too. Here's the ```main``` method: 
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

didymusne committed:
add level-1 minimal skeleton features - greet, echo, exit

didymusne committed:
add level-2 minimum skeleton features - add, list

didymusne committed:
add level-3 minimum skeleton features - mark as done

didymusne committed:
add level-4 minimum skeleton features - todos, events, deadlines

didymusne committed:
add input/output redirection technique to semiautomate testing of Duke

didymusne committed:
add error handling - made new DukeException class

didymusne committed:
add level-6 delete capabilities

didymusne committed:
reconfigured ENUM for Actions, and added counter in the delete message which I forgot to add in Level-6

didymusne committed:
Add save features

The tasks entered are now saved automatically whenever the task
list changes.

didymusne committed:
Add parsing of Date type and format date

Dates for deadline/ event are now accepted in 'yyyy-mm-dd'
format, and printed in 'MMM dd yyyy' format with error handling

didymusne committed:
Merge branch 'branch-Level-7'

This merge adds save I/O features to master.

didymusne committed:
Merge branch 'branch-Level-8'

This merge brings the Date parse functionality to master.

didymusne committed:
Add Ui class for OOP

The Ui class abstracts away print messages to the user

didymusne committed:
Add Storage class for OOP

The Storage class abstracts away save and loading of tasks.

didymusne committed:
Add Parser and Command classes for OOP

Parser and Command abstract away the interpretation of
user commands.

didymusne committed:
Make program more OOP

Use Ui, Storage, Parser, TaskList, Task, and DukeException
classes and their children to abstract away details
and follow the OOP paradigm.

didymusne committed:
Organise into Packages

Put all classes into packages.

didymusne committed:
Add JUnit Tests

Add JUnit tests to test the behavior of the code.

didymusne committed:
Add Jar - manifest file

didymusne committed:
Add find functionality

Give users a way to find a task by searching for a keyword.

didymusne committed:
Add JavaDoc comments

Add JavaDoc comments to the program

didymusne committed:
Tweak the code to comply with coding standard

Ensure that any new code is compliant.

didymusne committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDoc comments to the code.

didymusne committed:
Resolve merge conflict in TaskCommand

didymusne committed:
Resolve merge conflict in Parser

didymusne committed:
Merge branch 'add-gradle-support' into A-Gradle

Add gradle support.

didymusne committed:
Automate project builds using Gradle - run, build

Set up Gradle to build and run Duke.

didymusne committed:
Merge branch 'A-Gradle'

Automate project builds using gradle to build and run Duke with gradle.

didymusne committed:
Add GUI to Duke

didymusne committed:
Add GUI to Duke with JavaFX and fxml

didymusne committed:
Merge branch 'Level-10'

Add a GUI to Duke using JavaFX and fxml.

gok99 commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,29 @@
+import duke.Duke;
+import duke.task.TaskList;
+import duke.GUI;
+
```
imports here don't seem lexicographically ordered. i've omitted other import ordering related comments from this review

gok99 commented on src/main/java/duke/controller/DialogBox.java:
```diff
@@ -0,0 +1,99 @@
+
+    private DialogBox(String text, Image img) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBox.fxml"));
```
not a violation, but this is a particularly long statement (105 chars), coming close to the soft limit of 110

zacharylwy replied:
Sorry mate, I can barely find any violations in your code (and that is a good thing) soI will add on here that this line may need to be shortened. Keep up the good work

gok99 commented on src/main/java/duke/controller/DialogBox.java:
```diff
@@ -0,0 +1,99 @@
+     * of user.
+     */
+    public void modifyColorForError() {
+        this.dialog.setTextFill(color(1,1,1));
```
commas should be followed by whitespace

gok99 commented on src/main/java/duke/exception/DukeIncorrectTaskDescription.java:
```diff
@@ -0,0 +1,34 @@
+
+public class DukeIncorrectTaskDescription extends DukeIncorrectInputs {
+    private static final String PREFIX = "To log a";
+    private static final String SUFFIX = ", you have to log the " +
```
`+` should be on the next line 

(e.g):
```java
setText("Long line split"
        + "into two parts.");
```

(i've omitted comments for similar issues)

gok99 commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,20 @@
+import duke.GUI;
+import duke.task.TaskList;
+
+public class FindCommand extends Command {
```
Missing javadoc comment

gok99 commented on src/main/java/duke/GUI.java:
```diff
@@ -0,0 +1,112 @@
+     * @return formatted reply that is sent to the user.
+     */
+    private static String formatDukeMessage(String reply) {
+        return reply ;
```
Semicolon should not be preceded by whitespace

gok99 commented on src/main/java/duke/controller/MainWindow.java:
```diff
@@ -0,0 +1,93 @@
+        String input = userInput.getText().strip();
+        String response = " ";
+
+        DialogBox userDB = null;
```
should be named `userDb` instead:

(e.g.):

```java
exportHtmlSource();
openDvdPlayer();
```
instead of
```java
exportHTMLSource();
openDVDPlayer();
```

(similar comments omitted)

zacharylwy replied:
should be named userDb

gok99 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,82 @@
+package duke.task;
+
+import duke.exception.*;
```
wildcard imports should be avoided

gok99 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,82 @@
+           if (taskArr.length < 2) {
+               String[] emptyString = new String[2];
+               throw new DukeMissingTaskDescription(new Deadline(emptyString), new IllegalArgumentException());
+           } else if (taskArr[1].split("/by", 2).length < 2){
```
`{` should be preceded by whitespace

zacharylwy replied:
{ should have space before

gok99 reviewed (commented):
Nice bot! Looks mostly good to me, except for some minor style violations that I was able to spot.

zacharylwy reviewed (commented):
Looks great! Keep up the good work.

didymusne committed:
Update GUI to separate display picture from text

didymusne committed:
Fix bug in saving newly added tasks, add greeting

didymusne committed:
Make Find Command keyword case-insensitive

Keyword to find command is case sensitive.

A case insentive find is more user-friendly because
users cannot be expected to remember the exact case
of the keywords.

Let's update the keyword search algorithm to be case
insensitive.

didymusne committed:
Add assert statements

No assert statements.

Assert statements help document assumptions that should hold true.

didymusne committed:
Extract createFile method

Method to create file was not extracted.

This causes the code in Duke getResponse to be less readable.

Lets abstract away the details to create file if file does not
already exist.

didymusne committed:
Make happy path prominent for method getResponse

Happy path in method getResponse is not prominent.

The code will be less readable because of possible distraction.

Lets make the happy path more prominent to improve readability.

didymusne committed:
Extract method from parse method

Parse method has lower level details on implementation.

This reduces readability.

Lets abstract the details into new methods for so that the parse
method will only have step-by-step instructions.

didymusne committed:
Extract method from methods in Storage

Methods saveTasks and loadSavedTasks have lower level details.

This reduces readability.

Lets abstract these details into new methods.

didymusne committed:
Add SLAP to methods

didymusne committed:
Refactor code and extract methods

createTask method in Task class has low level implementation
details.

This reduces readability.

Lets abstract away the details into methods so that the code is
more readable.

didymusne committed:
Merge pull request #1 from didymental/branch-A-Assertions

Add assert statements

didymusne committed:
Merge branch 'master' into branch-A-CodeQuality

Update changes from master into branch-A-CodeQuality.

didymusne committed:
Merge pull request #2 from didymental/branch-A-CodeQuality

Merge branch-A-CodeQuality into master

didymusne committed:
Add sort tasks capabilities

Deadlines and events cannot be sorted by their date.

This hinders usability, especially since if users want to know
what is the nearest deadline or event.

Let's make the tasks comparable to sort them according to their
date.

didymusne committed:
Rename methods

didymusne committed:
Merge branch 'branch-C-Sort'

Add sorting capabilities to tasks so that users can sort their tasks
by the expected date.

didymusne committed:
Update README.md

didymusne committed:
Improve code quality and descriptive headers

didymusne committed:
Add functionality to close app window upon exit

didymusne committed:
Improve the GUI

didymusne committed:
Add representative screenshot of product

didymusne committed:
Add new screenshot of product

didymusne committed:
Remove screenshot

didymusne committed:
Add screenshot of product

didymusne committed:
Update README.md

didymusne committed:
Fix bugs from FILE PATH

didymusne committed:
Improve code quality

didymusne committed:
Make GUI unresizable

didymusne committed:
Update build.gradle to work on all os
